tests/test_optimizely_config.py

```diff
@@ -989,22 +1466,215 @@ def test_stringify_conditions(self):
+                      'audienceIds': ['0'], 'forcedVariations': {}, 'id': '1323241598'}
+
+        audience_conditions = experiment.get('audienceConditions')
+
+        audiences_map = {
+            '3468206642': 'us',
+        }
+
+        config = optimizely_config.OptimizelyConfig(
+            revision='101',
+            experiments_map={},
+            features_map={},
+            environment_key='TestEnvironmentKey',
+            attributes={},
+            events={},
+            audiences=None
+        )
+
+        config_service = optimizely_config.OptimizelyConfigService(config)
+
+        result = config_service.stringify_conditions(audience_conditions, audiences_map)
+
+        expected_result = '"us"'
+
+        self.assertEqual(result, expected_result)
+
+    def test_stringify_conditions_no_condition_word(self):
+        '''
+            Test to confirm list of ID's is handled as OR by stringify_conditions functions
+        '''
+
+        experiment = {'audienceConditions': ['3468206642', '3988293898'],
+                      'audienceIds': ['0'], 'forcedVariations': {}, 'id': '1323241598'}
+
+        audience_conditions = experiment.get('audienceConditions')
+
+        audiences_map = {
+            '3468206642': 'us',
+            '3988293898': 'female'
+        }
+
+        config = optimizely_config.OptimizelyConfig(
+            revision='101',
+            experiments_map={},
+            features_map={},
+            environment_key='TestEnvironmentKey',
+            attributes={},
+            events={},
+            audiences=None
+        )
+
+        config_service = optimizely_config.OptimizelyConfigService(config)
+
+        result = config_service.stringify_conditions(audience_conditions, audiences_map)
+
+        expected_result = '"us" OR "female"'
+
+        self.assertEqual(result, expected_result)
+
+    def test_stringify_conditions_and_or_and(self):
+        '''
+            Test to confirm [and[or][and]] is handled properly by stringify_conditions functions
+        '''
+
+        experiment = {'audienceConditions': ['and', ['or', '1', ['and', '2', '3']], ['and', '11', ['or', '12', '13']]],
+                      'audienceIds': ['0'], 'forcedVariations': {}, 'id': '1323241598'}
+
+        audience_conditions = experiment.get('audienceConditions')
+
+        audiences_map = {
+            '1': 'us',
+            '2': 'female',
+            '3': 'adult',
+            '11': 'fr',
+            '12': 'male',
+            '13': 'kid'
+        }
+
+        config = optimizely_config.OptimizelyConfig(
+            revision='101',
+            experiments_map={},
+            features_map={},
+            environment_key='TestEnvironmentKey',
+            attributes={},
+            events={},
+            audiences=None
+        )
+
+        config_service = optimizely_config.OptimizelyConfigService(config)
+
+        result = config_service.stringify_conditions(audience_conditions, audiences_map)
```

Comment:
Wondering if we can reuse this part, repeated the same for many tests.

Reply:
moved to a function to get the config and map so we can reuse this code.